src/sciline/pipeline.py

```diff
@@ -434,6 +433,18 @@ def __setitem__(self, key: Type[T], param: T) -> None:
             )
         self._set_provider(key, Provider.parameter(param))
 
+    def __getitem__(self, tp: Union[Type[T], Item[T]]) -> Provider:
```

Comment:
This makes `Provider` part of the public interface. So can you import it as well as ArgSpec, ProviderLocation, ProviderKind, ToProvider in the top level __init__.py? (ToProvider maybe should be exposed in typing instead)